src/runtime/pkg/containerd-shim-v2/service.go

```diff
@@ -155,40 +181,10 @@ type service struct {
 	pid uint32
 }
 
-func newCommand(ctx context.Context, id, containerdBinary, containerdAddress string) (*sysexec.Cmd, error) {
-	ns, err := namespaces.NamespaceRequired(ctx)
-	if err != nil {
-		return nil, err
-	}
-	self, err := os.Executable()
-	if err != nil {
-		return nil, err
-	}
-	cwd, err := os.Getwd()
-	if err != nil {
-		return nil, err
-	}
-	args := []string{
-		"-namespace", ns,
-		"-address", containerdAddress,
-		"-publish-binary", containerdBinary,
-		"-id", id,
-	}
-	opts := ctx.Value(cdshim.OptsKey{}).(cdshim.Opts)
-	if opts.Debug {
-		args = append(args, "-debug")
-	}
-	cmd := sysexec.Command(self, args...)
-	cmd.Dir = cwd
-
-	// Set the go max process to 2 in case the shim forks too much process
-	cmd.Env = append(os.Environ(), "GOMAXPROCS=2")
-
-	cmd.SysProcAttr = &syscall.SysProcAttr{
-		Setpgid: true,
-	}
-
-	return cmd, nil
+func (s *service) RegisterTTRPC(server *ttrpc.Server) error {
+	taskAPI.RegisterTaskService(server, s)
+	portForwardAPI.RegisterPortForwardService(server, s)
```

Comment:
Can we seperate out the port forward service addition from the shim service changes for making the shim a plugin?

Reply:
Ex, have a final "add port forward service" commit after the plumbing is complete -- I want to keep these changes as their own standalone commit:
- Make shim.TaskAPI a plugin
- Run shim using shimManager

Reply:
Done